src/messagingsession/MessagingSession.ts

```diff
@@ -9,6 +9,11 @@ export default interface MessagingSession {
    */
   start(): void;
 
+  /**
+   * Start a messaging session.
+   */
+  startAsync(): Promise<void>;
```

Comment:
- We have to make this optional method as we do not know if some builders are implementing the MessagingSession. If they are, then they have to implement this method and hence that would be a breaking change for them. 
- We would also need to update the 2.x README under the MessagingSession on using the `startAsync` method over `start` so that builders know why this is updated.